snewpdag/plugins/DtsCalculator.py

```diff
@@ -0,0 +1,117 @@
+"""
+DtsCalculator: compute neutrino time dts between each detector sending an alert.
+
+Input payload:
+    data[neutrino_time] = observed burst time for a detector (s,ns). For mc trials use
+    the DetectorTime.py plugin to generate this.
+
+Output payload:
+    dts: a dictionary of time differences. Keys are of form (det1, det2). The values are themselves dictionaries with
+    the following keys:
+    dt: time tuple (s,ns) of t1-t2
+    t1: neutrino time tuple (s,ns) of det1
+    t2: neutrino time tuple (s,ns) of det2
+"""
+
+import logging
+import numpy as np
+from snewpdag.dag import Node, lib
+
+class DtsCalculator(Node):
+        def __init__(self, **kwargs):
+            self.valid = [False] * 10  # flags indicating valid data from sources
+            self.t = [None] * 10  # neutrino time for each detector
+            self.dets_names = [None] * 10
+            self.dets_nu_times = [0.0] * 10
+            self.map = {}
```

Comment:
Does this fix the number of detectors to 10 max?

One could also use self.map to store the data (as it already does for the histories).  In this case it's indexed not by a number but by the detector name.